### PR TITLE
CCS-4249 sso enhancement

### PR DESCRIPTION
**Pull request by xdavidson**
This PR introduces the following changes:

- create a login servlet so that when users visit pantheon homepage, it's not automatically redirected to SSO
- create a logout filter that users can sign out the keycloak/SSO session if choose to.

### CODE REVIEW

File: pantheon-bundle/frontend/src/app/components/Chrome/Header/User.test.tsx [hunk 1]
```diff
@@ -13,11 +13,11 @@ describe("User tests", () => {
...
-  it("should render a Link component", () => {
-    const wrapper = mount(<Router><User {...mockStateUser} /></Router>)
-    const navLinks = wrapper.find(Link)
-    expect(navLinks.exists()).toBe(true)
-  })
+  // it("should render a Link component", () => {
+  //   const wrapper = mount(<Router><User {...mockStateUser} /></Router>)
+  //   const navLinks = wrapper.find(Link)
+  //   expect(navLinks.exists()).toBe(true)
+  // })
```
Comment (carlosmunoz) [inline pantheon-bundle/frontend/src/app/components/Chrome/Header/User.test.tsx]: Let's remove these code block comments

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/sso/LoginServlet.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * A login servlet that handles Keycloak or basic auth request
+ *
+ * @author Lisa Davidson
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                "sling.servlet.methods={GET, POST}",
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/auth/sso/LoginServlet.java:21]: minor: there is a `@SlingServlet` (or something like that) annotation that makes this a bit cleaner.

**Review by carlosmunoz — CHANGES_REQUESTED** [review]
nothing major. can't wait to see this released!

File: pantheon-bundle/frontend/src/app/components/Chrome/Header/User.test.tsx (hunk 1 above)
Comment (xdavidson) [reply pantheon-bundle/frontend/src/app/components/Chrome/Header/User.test.tsx]: done

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/sso/LoginServlet.java (hunk 2 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/auth/sso/LoginServlet.java]: updated!

**Review by carlosmunoz — APPROVED** [review]

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/sso/LoginServlet.java (hunk 2 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/auth/sso/LoginServlet.java]: according to this documentation, osgi annotation is preferred.
  The @SlingServlet annotation (evaluated by maven-scr-plugin, use only if you can neither use 1. nor 2.)
https://sling.apache.org/documentation/the-sling-engine/servlets.html

**Review by carlosmunoz — APPROVED** [review]

**Pull request merged**